### DISCUSSION TITLE
New Source: XML Source

### DISCUSSION THREAD

**Discussion by mcoirad** [airbytehq/airbyte#34967] [New Connector Request]
So have been considering using Airbyte as a tool for a new public governmental org, and one of the requirements is being able to handle XML files. For some reason had originally believed Airbyte had supported this, but as of now in the official docs it is listed as [an unsupported format. ](https://docs.airbyte.com/integrations/sources/file)

([For some reason, third party hosted docs listed it as supported.](https://www.restack.io/docs/airbyte-knowledge-airbyte-xml-integration) Wonder if they are AI generated or something. Thanks google.)

Noting that it has been brought up before here: https://github.com/airbytehq/airbyte/issues/5109 , but the requirements are out of date. In addition it seems the S3 connector is undergoing a refactor, so would probably want to wait for that to finish.

I am potentially interested at taking a stab at implementation, from what I understand that would require:

- Expand functionality of [/airbyte-integrations/connectors/source-file/source_file/client.py](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-file/source_file/client.py) to support XML, using pandas.read_xml
-- `pandas.read_xml` supports XLST for flattening XML documents into one table, potentially make that available as an option
- Add `xml` to list of supported formats in [/airbyte-integrations/connectors/source-file/source_file/spec.json](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-file/source_file/spec.json)
- Create unit tests as needed

Am I missing anything? Does this sound possible? Has this not been done before for any specific reason? Have not yet contributed to Airbyte as a developer so imagine there are more things to consider.

**Comment by natikgadzhi** [comment]
Hey! I know I'm late to the party. But, we're getting there. If there is a specific API you need to ingest as an example, I would love to work with you on getting XML decoder — we've expanded from just JSON to JSONL and friends, and I have a PR with CSV support.

**Comment by mcoirad** [comment]
Sorry, I also missed your reply. I am still interested in getting this done. If you can link me that PR I'll take a look and get started. Did my plan of action sound correct?

**Comment by natikgadzhi** [comment]
XML responses are already supported in Builder out of the box.

[a comment by an automated account is not shown]
